platform/kube/controller.go

```diff
@@ -420,15 +442,27 @@ func convertService(svc v1.Service) *model.Service {
 	}
 
 	return &model.Service{
-		Name:      svc.Name,
-		Namespace: svc.Namespace,
-		Ports:     ports,
-		Addresses: addrs,
+		Hostname: fmt.Sprintf("%s.%s.%s", svc.Name, svc.Namespace, ServiceSuffix),
+		Ports:    ports,
+		Address:  addr,
 		// TODO: empty set of service tags for now
 		Tags: nil,
 	}
 }
 
+// parseHostname is the inverse of hostname pattern from Kubernetes/Istio service translation
+func parseHostname(hostname string) (string, string, error) {
+	prefix := strings.TrimSuffix(hostname, "."+ServiceSuffix)
+	if len(prefix) < len(hostname) {
```

Comment:
nit: Consider reducing the indentation level here by returning early on errors. See https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow